mailpoet/lib/Automation/Integrations/MailPoet/Hooks/AutomationEditorLoadingHooks.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Hooks;
+
+use MailPoet\Automation\Engine\Data\Step;
+use MailPoet\Automation\Engine\Data\Workflow;
+use MailPoet\Automation\Engine\Hooks;
+use MailPoet\Automation\Engine\Storage\WorkflowStorage;
+use MailPoet\Newsletter\NewslettersRepository;
+use MailPoet\WP\Functions as WP;
+
+class AutomationEditorLoadingHooks {
+
+  /** @var WP */
+  private $wp;
+
+  /** @var WorkflowStorage  */
+  private $workflowStorage;
+
+  /** @var NewslettersRepository  */
+  private $newslettersRepository;
+
+  public function __construct(
+    WP $wp,
+    WorkflowStorage $workflowStorage,
+    NewslettersRepository $newslettersRepository
+  ) {
+    $this->wp = $wp;
+    $this->workflowStorage = $workflowStorage;
+    $this->newslettersRepository = $newslettersRepository;
+  }
+
+  public function init(): void {
+    $this->wp->addAction(Hooks::EDITOR_BEFORE_LOAD, [$this, 'beforeEditorLoad']);
+  }
+
+  public function beforeEditorLoad(int $workflowId): void {
+    $workflow = $this->workflowStorage->getWorkflow($workflowId);
+    if (!$workflow) {
+      return;
+    }
+    $this->disconnectEmptyEmailsFromSendEmailStep($workflow);
+  }
+
+  private function disconnectEmptyEmailsFromSendEmailStep(Workflow $workflow): void {
+    $sendEmailSteps = array_filter(
+      $workflow->getSteps(),
+      function(Step $step): bool {
+        return $step->getKey() === 'mailpoet:send-email';
+      }
+    );
+    foreach ($sendEmailSteps as $step) {
+      $emailId = $step->getArgs()['email_id'] ?? 0;
+      if (!$emailId) {
+        continue;
+      }
+      $newsletterEntity = $this->newslettersRepository->findOneById($emailId);
+      if ($newsletterEntity && $newsletterEntity->getBody() !== null) {
+        continue;
+      }
+      $updatedStep = new Step(
+        $step->getId(),
+        $step->getType(),
+        $step->getKey(),
+        array_merge($step->getArgs(), ['email_id' => 0]),
```

Comment:
That would be something, I am interested to hear @JanJakes as well. I actually prefer having a permanent `email_id` property which is either `null` or `0` by default. But I might miss the reasoning behind not having the property by default.

Reply:
@websupporter Sorry guys, I totally missed this was reassigned back to me.

Having/not having the email ID and having `0` vs `null` is rather a cosmetic issue, but it's true this it's not really compatible with what the step is doing (it doesn't exactly "undo" to the previous state without an email), so I agree with @johnolek it would make more sense to omit the `email` id in that case.

Note the validation as well, the email ID arg can be defined as `Builder::integer()->required()`. That doesn't allow for `null`, but in draft, this is not validated (only on activation). As for `0`, that seems a bit like a "hack" so I'd maybe avoid that, if possible. The `null` is fine, but at the moment it adds another state, it would make more sense only when combined with a real `null` default, such as `Builder::integer()->required()->default(null)` — and yeah, the default is invalid for the validation (which in this case is what we want).

Reply:
Addressed in 1389f6169